### PR TITLE
Modal refactoring for sandbox dialogues

### PR DESCRIPTION
**Pull request by burzum619**

### CODE REVIEW

File: webapp/src/main/webapp/src/app/admin/sandbox/modals/archive-sandbox.modal.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  archived: EventEmitter<SandboxConfiguration> = new EventEmitter();
+
+  constructor(private modal: BsModalRef, private service: SandboxService) {}
+
+  cancel() {
+    this.modal.hide();
+  }
+
+  archive() {
+    // this.service.archive(this.sandbox.code);
```
Comment (malaffoon) [inline webapp/src/main/webapp/src/app/admin/sandbox/modals/archive-sandbox.modal.ts]: TODO or lava code?

File: webapp/src/main/webapp/src/app/admin/sandbox/modals/reset-data.modal.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  resetData: EventEmitter<SandboxConfiguration> = new EventEmitter();
+
+  constructor(private modal: BsModalRef, private service: SandboxService) {}
+
+  cancel() {
+    this.modal.hide();
+  }
+
+  reset() {
+    // this.service.archive(this.sandbox.code);
```
Comment (malaffoon) [inline webapp/src/main/webapp/src/app/admin/sandbox/modals/reset-data.modal.ts]: This one is even more confusing (to me).

File: webapp/src/main/webapp/src/app/admin/sandbox/modals/archive-sandbox.modal.ts (hunk 1 above)
Comment (burzum619) [reply webapp/src/main/webapp/src/app/admin/sandbox/modals/archive-sandbox.modal.ts]: Yeah, let me add some TODOs. In fact we should probably revisit whether or not we're going to be supporting the "reset data" or "archive sandbox" functionalities.

File: webapp/src/main/webapp/src/app/admin/sandbox/modals/reset-data.modal.ts (hunk 2 above)
Comment (burzum619) [reply webapp/src/main/webapp/src/app/admin/sandbox/modals/reset-data.modal.ts]: The incorrect method call is a typo, but this needs a TODO as well

File: webapp/src/main/webapp/src/app/admin/sandbox/modals/delete-sandbox.modal.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { BsModalRef } from 'ngx-bootstrap';
+import { Component, EventEmitter } from '@angular/core';
+import { SandboxService } from '../sandbox.service';
+import { SandboxConfiguration } from '../sandbox-configuration';
+
+@Component({
+  selector: 'delete-sandbox-modal',
+  templateUrl: './delete-sandbox.modal.html'
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/admin/sandbox/modals/delete-sandbox.modal.ts]: for these i would add
```
changeDetection: ChangeDetectionStrategy.OnPush
```
and
```

ngOnDestroy(): void {
  myEventEmitter.complete();
}
```
to release subscriptions

File: webapp/src/main/webapp/src/app/admin/sandbox/sandbox.component.ts [hunk 4]
```diff
@@ -10,9 +15,78 @@ import { SandboxService } from './sandbox.service';
...
+        this.unsubscribe();
+      })
+    );
+  }
+
+  unsubscribe() {
+    this._modalSubscriptions.forEach(subscription =>
+      subscription.unsubscribe()
+    );
+    this._modalSubscriptions = [];
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/admin/sandbox/sandbox.component.ts]: i think we can remove this complexity if we go with the ngOnDestroy + complete() method

**Pull request merged**